File: java-app/src/main/java/com/mercadolibre/java_app/congrats/presenter/CongratsPresenter.java
package com.mercadolibre.java_app.congrats.presenter;

import com.mercadolibre.java_app.R;
import com.mercadolibre.java_app.commons.Purchase;
import com.mercadolibre.java_app.congrats.CongratsContract;

public class CongratsPresenter {
    private final CongratsContract.CongratsView congratsView;
    private final Purchase purchase;

    public CongratsPresenter(CongratsContract.CongratsView congratsView, Purchase purchase) {
        this.congratsView = congratsView;
        this.purchase = purchase;
    }

    public void setCongratsTexts() {
        showCongratsTexts();
        showShipping();
    }

    private void showShipping() {
        purchase.getShippingOption().showShippingPrices(congratsView);
    }

    private void showCongratsTexts() {
        congratsView.showTitle("Excelente compra!", R.color.green);
        congratsView.showFooter("Te avisaremos cuando esté en camino!");
    }
}

File: kt-app/src/main/java/com/mercadolibre/melikotlinmeetup/congrats/presenter/CongratsPresenter.kt
package com.mercadolibre.melikotlinmeetup.congrats.presenter

import com.mercadolibre.melikotlinmeetup.R
import com.mercadolibre.melikotlinmeetup.commons.Purchase
import com.mercadolibre.melikotlinmeetup.congrats.CongratsView

class CongratsPresenter(private val congratsView: CongratsView, private val purchase: Purchase) {
    fun setCongratsTexts() {
        showCongratsTexts()
        showShipping()
    }

    private fun showShipping() {
        purchase.shippingOption.showShippingPrices(congratsView)
    }

    private fun showCongratsTexts() {
        congratsView.showTitle("Excelente compra!", R.color.green)
        congratsView.showFooter("Te avisaremos cuando esté en camino!")
    }
}